cmd/maya-apiserver/cstor-operator/cspc/handler.go

```diff
@@ -137,6 +142,13 @@ func (c *Controller) syncCSPC(cspc *apis.CStorPoolCluster) error {
 		return nil
 	}
 
+	if !cspc.DeletionTimestamp.IsZero() {
+		// if returns error, we need to reque the request;
+		// else will return from sync method to avoid further reconcilation
+		// of object
+		return c.handleDeletion(cspc, openebsNameSpace)
```

Comment:
We are following not to return an error here but log and return nil to avoid queuing again that can cause log flooding.